x-pack/plugins/spaces/server/lib/spaces_client/spaces_client.ts

```diff
@@ -209,9 +209,9 @@ export class SpacesClient {
       throw Boom.badRequest('This Space cannot be deleted because it is reserved.');
     }
 
-    await repository.delete('space', id);
-
     await repository.deleteByNamespace(id);
+
+    await repository.delete('space', id);
```

Comment:
I swapped the order of these. My thinking is that if `deleteByNamespace` fails due to inline scripts not being executed, we shouldn't delete the space itself. This gives us a better UX; when a user is on the Space Management page or the Space Edit page, if the delete fails, the space is still there and the error describes why it failed.